pinot-controller/src/test/java/org/apache/pinot/controller/api/resources/PinotLogicalTableResourceTest.java

```diff
@@ -129,90 +133,191 @@ public void testCreateUpdateDeleteLogicalTables(String logicalTableName, List<St
   }
 
   @Test
-  public void testLogicalTableValidationTests()
+  public void testLogicalTableQuoteConfigValidation()
       throws IOException {
-    String addLogicalTableUrl = _controllerRequestURLBuilder.forLogicalTableCreate();
+    List<String> physicalTableNamesWithType = createHybridTables(List.of("test_table_1"));
+    LogicalTableConfig logicalTableConfig =
+        getDummyLogicalTableConfig(LOGICAL_TABLE_NAME, physicalTableNamesWithType, BROKER_TENANT);
+    logicalTableConfig.setQuotaConfig(new QuotaConfig("10G", "999"));
+    try {
+      ControllerTest.sendPostRequest(_addLogicalTableUrl, logicalTableConfig.toSingleLineJsonString(), getHeaders());
+      fail("Logical Table POST request should have failed");
+    } catch (IOException e) {
+      assertTrue(e.getMessage().contains("Reason: 'quota.storage' should not be set for logical table"),
+          e.getMessage());
+    }
```

Comment:
Can you also get and verify that the match ? Basically test the whole round trip

Reply:
With validation failure, the logical table creation fails so nothing to get and verify.

Do you mean, verify that table is not created or something else?